src/Voron/Impl/Journal/JournalReader.cs

```diff
@@ -293,25 +377,40 @@ public override string ToString()
 
         public void Dispose()
         {
-            OnDispose?.Invoke(this);
+            if (_encryptionBuffers != null) // Encryption enabled
+			{ 
+                foreach (var buffer in _encryptionBuffers)
+                    Win32MemoryProtectMethods.VirtualFree(buffer.Pointer, UIntPtr.Zero, Win32MemoryProtectMethods.FreeType.MEM_RELEASE);
```

Comment:
You just broke our Linux support, because this will try to call Win32 methods directly